### DISCUSSION TITLE
Prefer string over date/time?

### DISCUSSION THREAD

**Discussion by rmccu** [jongpie/NebulaLogger#380] [Q&A] [answered]
https://github.com/jongpie/NebulaLogger/blob/3b04138cdb22211796870ef49d8b3773465641e9/nebula-logger/core/main/log-management/classes/LogEntryEventHandler.cls#L158

Why does this code seem to prefer parsing the date/time from a String rather than using the Timestamp__c value, without needing to convert? Seems like this should be re-written to prefer not to convert/parse the date/time from string.

**Comment by jongpie** [comment, marked as answer]
This is by design, the string-based value is more accurate than the actual `Timestamp__c` datetime field - I have a comment [elsewhere in the codebase](https://github.com/jongpie/NebulaLogger/blob/3b04138cdb22211796870ef49d8b3773465641e9/nebula-logger/core/main/logger-engine/classes/LogEntryEventBuilder.cls#L673-L675) that provides a bit more context, but essentially, the issue is that Salesforce does not provide accurate timestamps for platform events within an Apex trigger (see screenshot below), so I use the string version of the timestamp because it will include accurate milliseconds when it's converted back into a datetime value.

![image](https://user-images.githubusercontent.com/1267157/193715044-66fb3ed1-9fb6-46e4-8a91-fa6323ee8ac8.png)

**Reply by rmccu** [reply]
Huh, wasn't aware of this, but good point. It might be worth adding a comment explaining this in the LogEntryEventHandler class, or at least a comment referencing this explanation in the LogEntryEventBuilder class.
